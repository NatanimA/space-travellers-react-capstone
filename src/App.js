import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Myprofile from './components/Myprofile';
import NavBar from './components/NavBar';
import Rockets from './components/rockets/Rockets';
import Missions from './components/missions/Missions';

function App() {
  return (
    <div className="App">
      <NavBar />
      <Routes>
        <Route exat path="/" element={<Rockets />} />
        <Route path="/missions" element={<Missions />} />
        <Route path="/profile" element={<Myprofile />} />
      </Routes>
    </div>
  );
}

export default App;
