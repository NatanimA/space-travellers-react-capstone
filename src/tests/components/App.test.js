import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import App from '../../App';

test('renders app component snapshot', () => {
  const component = render(
    <Provider store={store}>
      <MemoryRouter>
        <App />
      </MemoryRouter>
      ,
    </Provider>,
  );
  expect(component).toMatchSnapshot();
});
