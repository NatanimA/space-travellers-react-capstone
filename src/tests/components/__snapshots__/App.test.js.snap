// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <div>
            <img
              alt="Brand"
              class="brand-logo"
              src="brand.png"
            />
            <h1>
              Space Traveler's Hub
            </h1>
          </div>
          <nav>
            <ul>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                   Rockets
                </a>
              </li>
              <li
                class="nav-cutter"
              >
                <a
                  class=""
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li>
                <a
                  class="my-profile"
                  href="/profile"
                >
                  My profile
                </a>
              </li>
            </ul>
          </nav>
        </header>
        <section
          class="section-containers section-rockets"
        >
          <div
            class="rockets-container"
          />
        </section>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <div>
          <img
            alt="Brand"
            class="brand-logo"
            src="brand.png"
          />
          <h1>
            Space Traveler's Hub
          </h1>
        </div>
        <nav>
          <ul>
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                 Rockets
              </a>
            </li>
            <li
              class="nav-cutter"
            >
              <a
                class=""
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class="my-profile"
                href="/profile"
              >
                My profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <section
        class="section-containers section-rockets"
      >
        <div
          class="rockets-container"
        />
      </section>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
